modern/src/LanguageProvider.js

```diff
@@ -0,0 +1,18 @@
+import {LanguageContext} from "./LanguageContext";
+import React, {useState} from "react";
+import {defaultLanguage, setSelectedLanguage} from "./common/localization";
+
+const LanguageProvider = (props) => {
+    const [language, setLanguage] = useState(defaultLanguage);
+
+    const handleLanguageChange = (nextLanguage) => {
+        setSelectedLanguage(nextLanguage);
+        setLanguage(nextLanguage);
+    };
```

Comment:
Once we do the above, we probably won't need this anymore.

Reply:
I don't understand this comment.
Could you please explain how the code can be simplified?

Reply:
Once you move the code from `localization.js`, you won't need this method. You would just call `setLanguage`.